util/src/test/java/io/camunda/zeebe/util/sched/TestConcurrencyControl.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.util.sched;
+
+import io.camunda.zeebe.util.sched.future.ActorFuture;
+import io.camunda.zeebe.util.sched.future.TestActorFuture;
+import java.util.function.BiConsumer;
+
+/**
+ * Test implementation of {@code ConcurrencyControl}. The main goal is to use this in tests without
+ * starting the actor scheduler.
```

Comment:
:x: Please document the expected invariants and behavior. i.e. this only works with `TestActorFuture`, and will run any `Runnable` in submit inline in the current thread.